projects/smoke_mcp3427/src/main.c

```diff
@@ -0,0 +1,210 @@
+// A simple smoke test for MCP3427_AMP_GAIN_1.
+
+// Log readings whenever readings are available from the voltage/current sense MCP3427s.
+// Configurable items: I2C settings, MCP3427 device settings, which MCP3427s to be tested.
+// More detailed guide can be found here:
+// https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/1475543041/MCP3427+smoke+test+user+guide
+#include "mcp3427_adc.h"
+
+#include "event_queue.h"
+#include "gpio.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "wait.h"
+
+// Which channel on the MCP3427s to read from. By default channel 1 according to the schematics.
+#define SENSE_MCP3427_CHANNEL MCP3427_CHANNEL_1
+
+// MCP3427 device settings
+#define SMOKE_SAMPLE_RATE MCP3427_SAMPLE_RATE_12_BIT
+#define SMOKE_AMP_GAIN MCP3427_AMP_GAIN_1
+#define SMOKE_CONVERSION_MODE MCP3427_CONVERSION_MODE_ONE_SHOT
+
+#define MAX_NUM_MCP3427 7
+static SmokeMcp3427Data s_mcp3427_data[MAX_NUM_MCP3427];
```

Comment:
The `s_mcp3427_data` declaration has to be moved below the declaration of `SmokeMcp3427Data` for this to compile, the compiler hasn't seen the type declaration yet here.